File: android/src/main/java/com/getcapacitor/community/classes/results/StartResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;

public class StartResult implements Result {

    private final String peerID;

    public StartResult(String peerID) {
        this.peerID = peerID;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (peerID != null) result.put("peerID", peerID);

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Results/StartResult.swift
import Foundation
import Capacitor

@objc public class StartResult: NSObject, Result {
    let peerID: String?

    init(_ peerID: String?) {
        self.peerID = peerID
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        if let peerID = peerID {
            result["peerID"] = peerID
        }

        return result as AnyObject
    }
}
